yarn-project/acir-simulator/src/client/view_data_oracle.ts

```diff
@@ -113,6 +120,30 @@ export class ViewDataOracle extends TypedOracle {
     );
   }
 
+  /**
+   * Gets number of a block in which a given nullifier tree root was included.
+   * @param nullifierTreeRoot - The nullifier tree root to get the block number for.
+   * @returns The block number.
+   *
+   * TODO(#3564) - Nuke this oracle and inject the number directly to context
+   */
+  public async getNullifierRootBlockNumber(nullifierTreeRoot: Fr): Promise<number | undefined> {
```

Comment:
As before, not clear to me why it should be the nullifier root, but think it was just to have one?